localstack/services/kinesis/kinesis_listener.py

```diff
@@ -334,9 +334,6 @@ def subscribe_to_shard(data, headers):
     iterator = kinesis.get_shard_iterator(
         StreamName=stream_name, ShardId=data["ShardId"], ShardIteratorType=iter_type, **kwargs
     )["ShardIterator"]
-    data_needs_encoding = False
-    if "java" in headers.get("User-Agent", "").split(" ")[0]:
```

Comment:
Thanks for digging into this @SwatsonCodes ! 👍 

Looks like we introduced this switch in here for compatibility with the Kinesis Java SDK in [this PR](https://github.com/localstack/localstack/pull/3758). The original tests were introduced in our java-utils repo here: https://github.com/localstack/localstack-java-utils/pull/47/files - should we give it a short try to see if these Java tests are still passing with the changes?

Reply:
good thinking, I'll give it a shot.

Reply:
As discussed in person, been able to verify that these changes work against the `KinesisSchedulerTest` Java test. 👍 
```
mvn -Dtest=cloud.localstack.awssdkv2.KinesisSchedulerTest test
...
[INFO] BUILD SUCCESS
```
Not sure if our Java tests are currently actually covering the `SubscribeToShard` API - perhaps that is something we could add at some point..